src/login/logind.c

```diff
@@ -473,27 +443,85 @@ static int manager_enumerate_sessions(Manager *m) {
 
         FOREACH_DIRENT(de, d, return -errno) {
                 Session *s;
-                int k;
 
                 if (!dirent_is_file(de))
                         continue;
 
-                k = manager_add_session(m, de->d_name, &s);
-                if (k < 0) {
-                        RET_GATHER(r, log_warning_errno(k, "Failed to add session by filename %s, ignoring: %m", de->d_name));
+                r = manager_add_session(m, de->d_name, &s);
+                if (r < 0) {
+                        RET_GATHER(ret, log_warning_errno(r, "Failed to add session by filename %s, ignoring: %m", de->d_name));
                         continue;
                 }
 
-                session_add_to_gc_queue(s);
+                r = session_load(s);
+                if (r < 0)
+                        RET_GATHER(ret, log_warning_errno(r, "Failed to deserialize session '%s', ignoring: %m", s->id));
 
-                RET_GATHER(r, session_load(s));
+                session_add_to_gc_queue(s);
         }
 
-        /* We might be restarted and PID1 could have sent us back the session device fds we previously
-         * saved. */
-        (void) manager_attach_fds(m);
+        n = sd_listen_fds_with_names(true, &fdnames);
+        if (n < 0)
+                return log_error_errno(n, "Failed to acquire passed fd list: %m");
 
-        return r;
+        for (int i = 0; i < n; i++) {
```

Comment:
`FOREACH_ARRAY()`?

Reply:
`fdnames` is indexed by `i`. So I don't think it is useful here?

Reply:
Indeed. Sorry for the noise.